<launch>
  <node pkg="xqserial_server" type="xqserial_server" name="motor_driver"  respawn="true" respawn_delay="10">
    <param name="debug_flag" value="false"/>
    <param name="wheel_separation" value="0.36"/>
    <param name="wheel_radius" value="0.0825"/>
    <param name="max_speed" value="5.0"/>
    <param name="port" value="/dev/ttyUSB001"/>
    <param name="port_sonar" value="/dev/stm32Sonar"/>
    <param name="rot_dist" value="0.21"/>
    <param name="tran_dist" value="0.3"/>
    <param name="power_scale" value="1.0069"/>
    <param name="r_min" value="1.0"/>
    <remap from="cmd_vel" to="/cmd_vel" />
  </node>
  <node pkg="tf" type="static_transform_publisher" name="baselink_broadcaster" args="0 0 0.15 0 0 0 1 base_footprint base_link 20">
  </node>
  <node pkg="tf" type="static_transform_publisher" name="baselinkSonar_broadcaster1" args="0.3 -0.150 0.44 0 0 0  base_link sonar1 50"/>
  <node pkg="tf" type="static_transform_publisher" name="baselinkSonar_broadcaster2" args="0.3 0.150 0.44 0 0 0  base_link sonar2 50"/>
  <node pkg="tf" type="static_transform_publisher" name="baselinkSonar_broadcaster3" args="0.3 -0.150 0.64 0 0 0  base_link sonar3 50"/>
  <node pkg="tf" type="static_transform_publisher" name="baselinkSonar_broadcaster4" args="0.3 0.150 0.64 0 0 0  base_link sonar4 50"/>
  <node pkg="tf" type="static_transform_publisher" name="imulink_broadcaster" args="0.07 -0.03 0 0 0 0 1 base_link imu 20"/>
</launch>
